pkg/boot/linux.go

```diff
@@ -231,6 +301,37 @@ func loadLinuxImage(li *LinuxImage, logger ulog.Logger, verbose bool) (*LoadedLi
 	}, cleanup, nil
 }
 
+// saveLoadedLinuxImage marshals a LoadedLinuxImage to a json file.
+//
+// Marshalling it to a json versus a binary format, makes it readable
+// and easier to tamper in case when we need change a field or two
+// for experiments and debug.
+//
+// With that said, it is obvious that this saved info can then be
+// loaded by a later kexec for further execution from an already loaded
+// linux image and original load options.
+func saveLoadedLinuxImage(lli *LoadedLinuxImage, p string) error {
+	f, err := os.OpenFile(p, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0666)
+	if err != nil {
+		return err
+	}
+	out, err := json.Marshal(lli)
+	if err != nil {
+		return err
+	}
+	nw, err := f.Write(out)
+	if err != nil {
+		return err
+	}
+	if nw != len(out) {
+		return fmt.Errorf("written %d bytes, want %d bytes", nw, len(out))
+	}
+	if err := f.Sync(); err != nil {
```

Comment:
Close does a Sync, this is superfluous

Reply:
no, it does not call FSync syscall within -- it is actually quite contrary to that, one need make sure call FSync syscall before close to make sure in memory copy are flushed properly. 

See https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/os/file_posix.go;l=21